plugins/hls-refactor-plugin/src/Development/IDE/GHC/ExactPrint.hs

```diff
@@ -430,6 +434,32 @@ graftDecls dst decs0 = Graft $ \dflags a -> do
             | otherwise = DL.singleton (L src e) <> go rest
     modifyDeclsT (pure . DL.toList . go) a
 
+modifySmallestDeclWithM ::
+    forall a.
+    (HasDecls a) =>
+    (SrcSpan -> Bool) ->
+    (LHsDecl GhcPs -> TransformT (Either String) (Maybe [LHsDecl GhcPs])) ->
+    a ->
+    TransformT (Either String) a
+modifySmallestDeclWithM validSpan f a = do
```

Comment:
could you add Haddock docs for this and other new functions please?

Reply:
Added some comments on both